core/org.eclipse.cdt.core/utils/org/eclipse/cdt/utils/Cygwin.java

```diff
@@ -0,0 +1,175 @@
+/********************************************************************************
+ * Copyright (c) 2022 徐持恒 Xu Chiheng
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0.
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ ********************************************************************************/
+package org.eclipse.cdt.utils;
+
+import java.io.File;
+import java.util.Map;
+
+import org.eclipse.core.runtime.IPath;
+import org.eclipse.core.runtime.Path;
+import org.eclipse.core.runtime.Platform;
+
+// A collection of Cygwin-related utilities.
+public class Cygwin {
```

Comment:
This codes doesn't really belong here, there is already [Cygwin.java](https://github.com/eclipse-cdt/cdt/blob/main/core/org.eclipse.cdt.core/utils/org/eclipse/cdt/internal/core/Cygwin.java) that it should live in.

AFAICT this code is making a bunch of assumptions that should be deferred to cygwin to make, using tools like cygpath as done by org.eclipse.cdt.internal.core.Cygwin.runCygpath(String[])

For example, this code is making assumptions about cygwin mounts and where to find cygwin itself.